pkg/acs/calacs/lib/sub2d.c

```diff
@@ -0,0 +1,57 @@
+# include <math.h>
+# include "hstio.h"
+# include "hstcalerr.h"	/* SIZE_MISMATCH */
+
+/* Subtract the second SingleGroup from the first, leaving the
+   result in the first.  
+
+   (*a) -= (*b)
+
+   The science data arrays are subtracted; the error arrays are combined;
+   the data quality arrays are combined.
+
+   M.D. De La Pena: 05 June 2018
+   Created this file/routine from sub1d.c.
+   the data quality arrays are combined.  For all extensions, the entire 2D
+   arrays are used.
+*/
+int sub2d (SingleGroup *a, SingleGroup *b) {
```

Comment:
I would have made a generic 2d vector func that had the op func passed into it, in this instance the op is the sub.

Reply:
Multi flavours or wrappers may be wanted/needed to deal with the DQ masks.

Reply:
No go.  This issue balloons into a lot of variations.